warehouse/oidc/forms/activestate.py

```diff
@@ -0,0 +1,245 @@
+            ),
+        ]
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _lookup_organization(self, org_url_name: str) -> None:
+        """Make gql API call to the ActiveState API to check if the organization
+        exists"""
+        try:
+            response = requests.post(
+                f"{ACTIVESTATE_GRAPHQL_API_URL}",
+                json={
+                    "query": GRAPHQL_GET_ORGANIZATION,
+                    "variables": {"orgname": org_url_name},
+                },
+                timeout=TIMEOUT,
+            )
+            response.raise_for_status()
+        except requests.HTTPError:
+            sentry_sdk.capture_message(
+                f"Unexpected error from ActiveState organization lookup: {response.content!r}"  # noqa
+            )
+            raise wtforms.validators.ValidationError(
+                _("Unexpected error from ActiveState. Try again")
+            )
+        except requests.ConnectionError:
+            sentry_sdk.capture_message(
+                "Connection error from ActiveState organization lookup API (possibly offline)"  # noqa
+            )
+            raise wtforms.validators.ValidationError(
+                _(
+                    "Unexpected connection error from ActiveState. "
+                    "Try again in a few minutes"
+                )
+            )
+        except requests.Timeout:
+            sentry_sdk.capture_message(
+                "Timeout from ActiveState organization lookup API (possibly offline)"
+            )
+            raise wtforms.validators.ValidationError(
+                _("Unexpected timeout from ActiveState. Try again in a few minutes")
+            )
+        # Graphql reports it's errors within the body of the 200 response
+        try:
+            response_json = response.json()
+            errors = response_json.get("errors")
+            if errors:
+                sentry_sdk.capture_message(
+                    f"Unexpected error from ActiveState organization lookup: {errors}"  # noqa
+                )
+                raise wtforms.validators.ValidationError(
+                    _("Unexpected error from ActiveState. Try again")
+                )
+
+            if response_json.get("data") and not response_json.get("data").get(
+                "organizations"
+            ):
+                raise wtforms.validators.ValidationError(
+                    _("ActiveState organization not found")
+                )
+        except requests.exceptions.JSONDecodeError:
+            sentry_sdk.capture_message(
+                f"Unexpected error from ActiveState organization lookup: {response.content!r}"  # noqa
+            )
+            raise wtforms.validators.ValidationError(
+                _("Unexpected error from ActiveState. Try again")
+            )
+
+    def validate_organization(self, field):
+        self._lookup_organization(field.data)
```

Comment:
Any reason not to just put the body of this function here instead?

Reply:
No dice on this.  Some step in the code path of looking up validation functions based on the field name causes stubbing not to work.  When I try to stub `validate_organization`, the original code runs instead of the stubbed function.  If I keep `_lookup_organization` the stub works.